core/src/main/java/com/linecorp/armeria/client/DefaultEventLoopScheduler.java

```diff
@@ -247,11 +248,14 @@ private void cleanup() {
 
         for (final Iterator<AbstractEventLoopState> i = states.values().iterator(); i.hasNext();) {
             final AbstractEventLoopState state = i.next();
+            final ReentrantLock lock = new ReentrantLock();
```

Comment:
I'm not sure this `lock` can protect `state` from multiple threads access.
Should we promote `lock` to a member variable?

Reply:
Oops, your right 😓